### ISSUE TITLE
Rule proposal: Disallow `declare` for exports

### ISSUE THREAD

**Issue by sindresorhus** [typescript-eslint/typescript-eslint#3016]
Using `declare` with `export` is moot.

## Fail

```ts
export declare function foo(): Promise<void>;
```

## Pass

```ts
export function foo(): Promise<void>;
```

**Comment by bradzacher** [comment]
> Using declare with export is moot.

Could you please explain what you mean by this?

Without `declare` your "pass" case raises a TS error: `Function implementation is missing or not immediately following the declaration. (2391)`.

**Comment by sindresorhus** [comment]
Both `tsc --noEmit index.d.ts` and [`tsd`](https://github.com/SamVerschueren/tsd) passed for me, so assumed it was moot.
